set.mm

```diff
@@ -369106,6 +369106,95 @@ varying definitions (some start from 0, others start from 1), but
        renamer theorem like ~ cbval . The section
        ~ mmtheorems32.html#mm3146s also describes the
        metatheorem that underlies this.
+       </p>
+
+       <p>
+       Standard Metamath verifiers do not distinguish between axioms and
+       definitions (both are $a statements).
+       In practice, we require that definitions (1) not be creative
+       (a definition should not allow an expression
+       that previously qualified as a wff but was not provable
+       to become provable) and be eliminable
+       (there should exist an algorithmic method for converting any
+       expression using the definition into
+       a logically equivalent expression that previously qualified as a wff).
+       To ensure this, we have additional rules on almost all definitions
+       ($a statements with a label that does not begin with ax-).
+       These additional rules are not applied in a few cases where they
+       are too strict ( ~ df-bi , ~ df-clab , ~ df-cleq , and ~ df-clel );
+       see those definitions for more information.
+       These additional rules for definitions are checked by at least
+       mmj2's definition check (see
+       <A HREF="https://github.com/digama0/mmj2/blob/master/"
+       >mmj2 master</a> file mmj2jar/macros/definitionCheck.js).
+       This definition check relies on the database being very much like
+       set.mm, down to the names of certain constants and types, so it
+       cannot apply to all Metamath databases... but it is useful in set.mm.
+       In this definition check, a $a-statement with a given label and
+       typecode ` |- ` passes the test if and only if it
+       respects the following rules (these rules imply that we have
+       an unambiguous tree parse, which is also checked):
+       </p>
+
+       <ol>
+       <li>The expression must be a biconditional or an equality (i.e. its
+           root-symbol must be ` <-> ` or ` = `).
+           If the proposed definition passes this first rule, we then
+           define its definiendum as its left hand side (LHS) and
+           its definiens as its right hand side (RHS).
+           We define the *defined symbol* as the root-symbol of the LHS.
+           We define a *dummy variable* as a variable occurring
+           in the RHS but not in the LHS.
+           Note that the "root-symbol" is the root of the considered tree;
+           it need not correspond to a single token in the database
+           (e.g., see ~ w3o or ~ wsb ).
+       <li>The defined symbol should not occur in an earlier
+           $a-statement with typecode ` |- ` or in any another place
+           in this labelled expression.
+       <li>No two variables occurring in the LHS should share a
+           disjoint variable (DV) condition.
+           Without this rule, you can define something like
```

Comment:
I agree with Benoit that the justification for the rules should come after the list. I'm not sure if we even need the example here, but please rename Bar to Foo if so.

Reply:
I think it's better where it is, but 2 people disagree with me & I don't care *that* much. So I'll move it.  I'll rename the example to foo.